sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/api/ApiExtensions.kt

```diff
@@ -482,19 +493,28 @@ fun PsiElement?.getType(bindingContext: BindingContext) =
 fun PsiElement?.determineType(bindingContext: BindingContext): KotlinType? =
     this?.let {
         when (this) {
-            is KtCallExpression -> getResolvedCall(bindingContext)?.getReturnType()
+            is KtCallExpression ->   getResolvedCall(bindingContext)?.resultingDescriptor?.returnType
             is KtParameter -> determineType(bindingContext)
             is KtTypeReference -> determineType(bindingContext)
             is KtProperty -> determineType(bindingContext)
             is KtDotQualifiedExpression -> resolveReferenceTarget(bindingContext).determineType()
             is KtReferenceExpression -> bindingContext.get(BindingContext.REFERENCE_TARGET, this).determineType()
             is KtFunction -> bindingContext.get(BindingContext.DECLARATION_TO_DESCRIPTOR, this).determineType()
             is KtClass -> bindingContext.get(BindingContext.DECLARATION_TO_DESCRIPTOR, this).determineType()
-            else -> bindingContext.get(BindingContext.VARIABLE, it)?.type
+            is KtExpression -> this.getKotlinTypeForComparison(bindingContext)
+            is KtValueArgument -> this.getArgumentExpression()?.predictRuntimeValueExpression(bindingContext)?.determineType(bindingContext)!!
+            else -> null
         }
 
     }
 
+fun KotlinType.isSupertype(other: KotlinType): Boolean {
+    var otherSuperTypes = other.supertypes()
+    if (otherSuperTypes.size == 1 && otherSuperTypes.any { it.toString() == ANY_TYPE }) return false
+    if (otherSuperTypes.any { it == this }) return true
+    return otherSuperTypes.any { isSupertype(it) }
+}
```

Comment:
Oops, didn't notice I could use that method! Looks good for the use I need, one thing that I think I should include for my case is to avoid returning `true` for cases like:
` Any.isSuperTypeOf(Any)` 
since I am already checking in the check for the argument to be different from the collection key, in order to raise an issue. Also it doesn't look right to return true when stating if a type is a supertype of itself. What do you think?